### PR TITLE
feat: better backtrack

### PR DESCRIPTION
**Pull request by larryngton2**
### added
- render modes
- chance
- random nextbacktrackdelay
- random backtrackdelay

### todo
- [x] playermodel render mode
- [x] refactor my shitcode
- [x] more settings

### CODE REVIEW

**Comment by ghost** [comment]
want something in client
make shit code for it
force izuna to recode it

**Comment by larryngton2** [comment]
> want something in client make shit code for it force izuna to recode it

99% larryman
1% eyezuna

(with the refactoring its 70% larryman 30% eyezuna)

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 1]
```diff
@@ -46,15 +49,22 @@ import net.minecraft.util.math.Vec3d
...
     private val range by floatRange("Range", 1f..3f, 0f..6f)
-    private val delay by int("Delay", 100, 0..1000, "ms").apply { tagBy(this) }
-    private val boxColor by color("BoxColor", Color4b(36, 32, 147, 87))
+    private val delay by intRange("Delay", 100..150, 0..1000, "ms")
+    private val nextBacktrackDelay by intRange("NextBacktrackDelay", 0..10, 0..2000, "ms")
+    private val chance by float("Chance", 50f, 0f..100f, "%")
+    private val renderMode = choices("RenderMode", Box, arrayOf(Box, Wireframe, None))
 
     private val packetQueue = LinkedHashSet<DelayData>()
+    private var delayForNextBacktrack = 0L
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: Use the `Chronometer` utility for that please

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 2]
```diff
@@ -46,15 +49,22 @@ import net.minecraft.util.math.Vec3d
...
 
     private var target: Entity? = null
     private var position: TrackedPosition? = null
 
     val packetHandler = handler<PacketEvent> {
+        if (packetQueue.isNotEmpty()) {
+            delayForNextBacktrack = System.currentTimeMillis() + nextBacktrackDelay.random()
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: s.a.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt [hunk 3]
```diff
@@ -225,9 +266,13 @@ object ModuleBacktrack : Module("Backtrack", Category.COMBAT) {
...
         enabled && packetQueue.isNotEmpty()
 
-    private fun shouldConsiderAsEnemy(target: Entity) =
-        target.shouldBeAttacked() && target.boxedDistanceTo(player) in range && player.age > 10
+    private fun shouldBacktrack(target: Entity) =
+        target.shouldBeAttacked() &&
+            target.boxedDistanceTo(player) in range &&
+            player.age > 10 &&
+            Math.random() * 100 < chance &&
+            System.currentTimeMillis() >= delayForNextBacktrack
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt]: s.a.

**Review by superblaubeere27 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt (hunk 2 above)
Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleBacktrack.kt:65]: can you translate to simple english

**Review by EclipsesDev — COMMENTED** [review]
Is this even needed?

**Comment by EclipsesDev** [comment]
no wait wrong one xd

**Comment by larryngton2** [comment]
> no wait wrong one xd

well which is the correct one

**Comment by EclipsesDev** [comment]
test @larryngton2

**Comment by larryngton2** [comment]
> test @larryngton2

tested successfully

**Review by 1zun4 — APPROVED** [review]
Code looks good but I have not tested.

**Pull request merged**